src/tests/subcommand_test.go

```diff
@@ -0,0 +1,77 @@
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+// runAndAssertSubcommand executes a subcommand of the "semantifly" command with the provided arguments
+// and asserts that the output contains the specified assertion statement.
+//
+// Parameters:
+//   - subcommand (string): The subcommand to execute.
+//   - assertStatement (string): The expected substring to be present in the command output.
+//   - args ([]string): Additional arguments to pass to the subcommand.
+func runAndAssertSubcommand(subcommand string, assertStatement string, args []string) error {
+	allArgs := append([]string{subcommand}, args...)
+	cmd := exec.Command("semantifly", allArgs...)
+
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("Command execution failed: %v\nStderr: %s", err, stderr.String())
+	}
+
+	output := stdout.String()
+
+	// Assertion for command output
+	if !strings.Contains(output, assertStatement) {
+		return fmt.Errorf("Expected output to contain %s, but got: %s", assertStatement, output)
+	}
+
+	return nil
+}
+
+func TestCommandRun(t *testing.T) {
```

Comment:
Please run setup.sh from within here so that our unit tests remain hermetic and runnable directly after cloning the repo

Reply:
Added